### ISSUE TITLE
[Coveralls] Add test cases to improve the coveralls percentage

### ISSUE THREAD

**Issue by eswarasai** [MARKETProtocol/Dapp#61]

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.3 ETH (202.18 USD @ $673.92/ETH) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/issue/MARKETProtocol/Dapp/61).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/issue/MARKETProtocol/Dapp/61)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5452.39 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.3 ETH (208.66 USD @ $695.55/ETH) funding by__: 
 1. [@MichelleKwa](https://gitcoin.co/profile/MichelleKwa) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder (@pelsasser) and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/MARKETProtocol/Dapp/61)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5571.7 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by pelsasser** [comment]
Hello @MichelleKwa - looking forward to your work! thanks for the interest in our project!

**Comment by perfectmak** [comment]
Hi @MichelleKwa one place that lacks test coverage are the redux actions. You might want to start from there.

**Comment by eswarasai** [comment]
@MichelleKwa -- Before jumping directly into fixing the tests, it'd be helpful for us to understand your plan of action on how you'd tackle this and also a list of what all tests that are needed to be added/fixed. Thanks.

**Comment by cryptomental** [comment]
@MichelleKwa and @saralarambhapuri  I would also like to understand what is your plan to tackle this bounty as learning how to correctly implement those tests interests me a lot.

**Comment by pelsasser** [comment]
@MichelleKwa @saralarambhapuri - any updates to progress here?

**Comment by pelsasser** [comment]
@MichelleKwa @saralarambhapuri - are you guys still working on this? Do we need to get some more help?  Maybe we can reach out to the gitcoin community for some help.  I can add some ETH to get some more manpower behind this.

**Comment by saralarambhapuri** [comment]
I need some help in the sample account and coinbase flow. I am currently stuck there

**Comment by saralarambhapuri** [comment]
Only after I get to know how that works, I can share the estimate

**Comment by pelsasser** [comment]
@saralarambhapuri - what is the plan for getting help with that? Do you need assistance in tracking down someone that can help? Maybe we can post a very specific issue that we can post to gitcoin for someone to resolve that knows how to address this?  Can you create the issue perhaps?

@eswarasai - thoughts on this plan?

**Comment by saralarambhapuri** [comment]
I need someone to help me get the coinbase data and create test account.

Creating a separate issue would help

@eswarasai do you agree?

**Comment by pelsasser** [comment]
@saralarambhapuri okay - please create the issue with as much details as possible and I can get it staked

**Comment by eswarasai** [comment]
@pelsasser -- I agree. Maybe @cryptomental or @perfectmak can guide us with setting up the basic framework to write tests for getting contracts from Rinkeby maybe? I'll also try to spend sometime on this and see how we can tackle this effectively.

**Comment by cryptomental** [comment]
@eswarasai , @saralarambhapuri the most difficult part in this task is to get first redux action test case working. I would say it is 70-80% of the overall effort.

**Comment by pelsasser** [comment]
@saralarambhapuri - any progress on getting a new issue created with more description of the problems you are facing?

**Comment by perfectmak** [comment]
I submitted a PR to provide tests for actions behaviors. It can be found here: #107 
I refactored actions to be initialized with their dependencies as inputs so it would easier to test.

Also, I noticed that most new contributions to the codebase don't have their components tested. We should endeavor to be adding some unit tests to every PR submitted.

**Comment by gitcoinbot** [comment]
__The funding of 0.3 ETH (154.18 USD @ $513.94/ETH) attached to this issue has been **killed** by the bounty submitter__

 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $3421.98 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by weberV** [comment]
Issue Status: 1. Open 2. **Expired** 

<hr>

__This issue now has a funding of 0.3 ETH (207.63 USD @ $692.1/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](http://localhost:8000/issue/MARKETProtocol/Dapp/61/118).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $1,512.27 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
